src/wallet/wallet.cpp

```diff
@@ -3908,6 +3894,15 @@ bool CWallet::ApplyMigrationData(MigrationData& data, bilingual_str& error)
             if (data.watchonly_wallet) {
                 LOCK(data.watchonly_wallet->cs_wallet);
                 if (data.watchonly_wallet->IsMine(*wtx->tx) || data.watchonly_wallet->IsFromMe(*wtx->tx)) {
+                    // Need to update WalletTx state before moving the
+                    // transaction from the main wallet to the watchonly wallet,
+                    // because the main wallet is an offline wallet created with
+                    // a dummy context, and doesn't know heights of confirmed
+                    // and conflicted blocks (values are set to -1). Calling
+                    // updateState will look up missing height values so the
+                    // watchonly wallet can function correctly after the
+                    // migration without needing to be reloaded.
+                    wtx->updateState(data.watchonly_wallet->chain());
                     // Add to watchonly wallet
                     if (!data.watchonly_wallet->AddToWallet(wtx->tx, wtx->m_state)) {
```

Comment:
not really for this PR but shouldn't we also transfer the other wtx members too?
Like `mapValue` (for the "replaced_by_txid" and "replaces_txid"), `nTimeReceived`, `nTimeSmart`, `fFromMe`.

Reply:
nvm, just saw #28609 implementing it.